### ISSUE TITLE
Geneformer

### ISSUE THREAD

**Issue by laraPPr** [vscentrum/vsc-software-stack#472]
* link to support ticket: [#2024120560001104](https://otrsdict.ugent.be/otrs/index.pl?Action=AgentTicketZoom;TicketID=178541)
* website: https://huggingface.co/ctheodoris/Geneformer
* installation docs: https://geneformer.readthedocs.io/en/latest/index.html
* toolchain: `foss/2023a`
* easyblock to use: `PythonBundle`
* required dependencies:
  * [ ] ...
* optional dependencies:
  * [ ] ...
* notes:
  * ...
* effort: *(TBD)*
* other install methods
  * conda: yes (link?) / no
  * container image: yes (link?) / no
  * pre-built binaries (RHEL8 Linux x86_64): yes (link?) / no
  * easyconfig outside EasyBuild: yes / no

**Comment by PetrKralCZ** [comment]
- https://github.com/easybuilders/easybuild-easyconfigs/pull/21994

**Comment by boegel** [comment]
@PetrKralCZ Can you summarize here what the problem is with the draft easyconfig for Geneformer?

**Comment by boegel** [comment]
@PetrKralCZ We should look into implementing support for `source_dependencies`, so we can make sure that `git-lfs` is available when the source tarball for `Geneformer` is being created. If not, the `model.safetensors` file will not be the actual contents of the file, but a placeholder with metadata that tells `git-lfs` where to get the actual file...

Maybe it's better to use `source_deps` as name for this easyconfig parameter, since we'll soon start migrating towards a more consistent set of easyconfig parameter names, see also https://github.com/easybuilders/easybuild-framework/issues/4464

Implementing support for `source_deps` can be done by:

* adding a new general easyconfig parameter `source_deps` to `easybuild/framework/easyconfig/default.py`, see the "`DEPENDENCIES easyconfig parameters`" section around line 168;
* picking up on `source_deps` in the `fetch_step` method in `easybuild/framework/easyblock.py`, something like:

```python
def fetch_step(...):

    source_deps = self.cfg['source_deps']
    pre_fetch_env = None
    # load modules for source dependencies (if any)
    if source_deps:
        pre_fetch_env = copy.deepcopy(os.environ)
        source_deps_mod_names = ...
        self.modules_tool.load(source_deps_mod_names)

    ...  # current body of fetch_step

    # restore environment (unload source dependency modules)
    if pre_fetch_env:
        restore_env(pre_fetch_env)

```

To facilitate getting the module names for the source dependencies, it's probably easiest to update the `parse` method in `easybuild/framework/easyconfig/easyconfig.py` to also parse `source_deps`, as is already done for `dependencies`, `builddependencies`, etc.

Do let me know if you need any further help with this!

**Comment by boegel** [comment]
Temporary workaround to get `Geneformer` installed while support for `source_deps` is not installed yet:
```
eb git-lfs-3.5.1.eb -r
module load git-lfs/3.5.1
eb --fetch --from-pr 21994
eb --from-pr 21994 -r
```

**Comment by boegel** [comment]
PR for CUDA version:
* https://github.com/easybuilders/easybuild-easyconfigs/pull/22200

**Comment by boegel** [comment]
@PetrKralCZ How much time did you spend on this more-or-less one prior to looking into implementing support for `source_deps` in framework?

**Comment by PetrKralCZ** [comment]
I think it took 2 days.

**Comment by boegel** [comment]
PR to add support for `source_deps`:
* https://github.com/easybuilders/easybuild-framework/pull/4766

**Comment by pavelToman** [comment]
PR for latest version of Geneformer with help of source_deps:
- https://github.com/easybuilders/easybuild-easyconfigs/pull/26813

**Comment by boegel** [comment]
@pavelToman Can you make an estimate of how much time (# hours) this request took you?

**Comment by pavelToman** [comment]
Took 4.5h
